src/udev/udev-builtin-path_id.c

```diff
@@ -256,11 +256,17 @@ static sd_device *handle_scsi_iscsi(sd_device *parent, char **path) {
 static sd_device *handle_scsi_ata(sd_device *parent, char **path) {
         sd_device *targetdev, *target_parent;
         _cleanup_(sd_device_unrefp) sd_device *atadev = NULL;
-        const char *port_no, *sysname;
+        const char *port_no, *sysname, *name;
+        int host, bus, target, lun;
 
         assert(parent);
         assert(path);
 
+        if (sd_device_get_sysname(parent, &name) < 0)
+                return NULL;
+        if (sscanf(name, "%d:%d:%d:%d", &host, &bus, &target, &lun) != 4)
```

Comment:
please use the correct types, i doubt these are actually signed values, are they?

Reply:
In other places, they are all defined as signed variables. Changing them to unsigned makes
sense to me.